modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/asset/display/page/portlet/ObjectEntryFolderDisplayPageFriendlyURLResolver.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.object.web.internal.asset.display.page.portlet;
+
+import com.liferay.asset.display.page.portlet.BaseAssetDisplayPageFriendlyURLResolver;
+import com.liferay.layout.display.page.LayoutDisplayPageProvider;
+import com.liferay.object.model.ObjectEntryFolder;
+import com.liferay.object.service.ObjectEntryFolderLocalService;
+import com.liferay.petra.string.CharPool;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.feature.flag.FeatureFlagManagerUtil;
+import com.liferay.portal.kernel.portlet.FriendlyURLResolver;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Guilherme Camacho
+ */
+@Component(service = FriendlyURLResolver.class)
+public class ObjectEntryFolderDisplayPageFriendlyURLResolver
+	extends BaseAssetDisplayPageFriendlyURLResolver {
+
+	@Override
+	public String getDefaultURLSeparator() {
+		return "/folder/";
+	}
+
+	@Override
+	public String getKey() {
+		return ObjectEntryFolder.class.getName();
+	}
+
+	@Override
+	public boolean isURLSeparatorConfigurable() {
+		return true;
+	}
+
+	@Override
+	protected LayoutDisplayPageProvider<?> getLayoutDisplayPageProvider(
+			String friendlyURL)
+		throws PortalException {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-21926")) {
```

Comment:
We are using LPD-17564 as FF for CMS in order to avoid having a ton of FF during the development.

In some specific cases we could use a different one, but this case seems to require LPD-17564.